### PR TITLE
[release/3.0] Use Microsoft.NETCore.App.Internal for runtime version

### PR DESCRIPTION
**Pull request by mmitche**
For stable builds, core-setup is now publishing its artifacts to a suffixed directory (e.g. 3.0.1-servicing-19510-13) instead of 3.0.1. This ensures we don't have to overwrite outputs when we rebuild stable versions. Within that directory, it publishes the same set of files with the final file names as well as suffixed file names:
- https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.0.1-servicing-19510-13/dotnet-apphost-pack-3.0.1-win-x64.msi
- https://dotnetcli.blob.core.windows.net/dotnet/Runtime/3.0.1-servicing-19510-13/dotnet-apphost-pack-3.0.1-servicing-19510-13-win-x64.msi

Downstream repos should install the runtime using the full suffixed version.

### CODE REVIEW

**Comment by mmitche** [comment]
Ugh, dotnet-install is making assumptions about the extraction directory it attempts to find. I think some tweaks may be necessary there.

**Comment by dougbu** [comment]
@mmitche all of `dotnet` makes assumptions about the names of the 3 shared runtimes. If Microsoft.NETCore.App.Runtime and Microsoft.NETCore.App.Internal packages have different internal layouts, nothing is going to work. (Arcade is using `dotnet-install.ps1` to add folders with newer runtime bits where the SDK expects them.) Might be able to move enough around for things to limp along but it seems perilous.

**Comment by mmitche** [comment]
> @mmitche all of `dotnet` makes assumptions about the names of the 3 shared runtimes. If Microsoft.NETCore.App.Runtime and Microsoft.NETCore.App.Internal packages have different internal layouts, nothing is going to work. (Arcade is using `dotnet-install.ps1` to add folders with newer runtime bits where the SDK expects them.) Might be able to move enough around for things to limp along but it seems perilous.

THis is just about the version number.  core-setup is making two copies of the final 3.0.1 bits, one with 3.0.1 file names, and one with 3.0.1-servicing-blah file names. It's placing those both in a final blob storage folder with 3.0.1-servicing so there is no overwriting.  This change is just so that we can find the exact 3.0.1 version of the runtime we want, since dotnet-install will construct the path based on the version number.

The issue is that at the very end of the script, it goes and checks whether the netcore.app share framework ended up in a 3.0.1-servicing-blah directory, which of course, since it is a final branded runtime, it did not.  It ended up in 3.0.1.

**Comment by mmitche** [comment]
This is blocked on @johnbeisner determining how the dotnet-install script can work properly in situations where the zip contents don't match the zip name

**Comment by mmitche** [comment]
@dougbu @wtgodbe Directly related to what we were talking about today. Can you take a look?

File: eng/Versions.props [hunk 1]
```diff
@@ -53,7 +53,8 @@
   <PropertyGroup Label="Automated">
     <!-- Packages from dotnet/core-setup -->
     <MicrosoftNETCoreAppRefPackageVersion>3.0.0</MicrosoftNETCoreAppRefPackageVersion>
-    <MicrosoftNETCoreAppRuntimewinx64PackageVersion>3.0.2-servicing-19576-08</MicrosoftNETCoreAppRuntimewinx64PackageVersion>
+    <MicrosoftNETCoreAppInternalVersion>3.0.2-servicing-19604-01</MicrosoftNETCoreAppInternalVersion>
```
Comment (wtgodbe) [inline eng/Versions.props]: > error : Missing version variable 'MicrosoftNETCoreAppInternalPackageVersion' in the 'Automated' property group in D:\a\1\s\/eng/Versions.props

This should be `MicrosoftNETCoreAppInternalPackageVersion` to satisfy codeCheck

**Pull request merged**